modules/compliance-tests/src/smithy4s/compliancetests/internals/package.scala

```diff
@@ -62,4 +63,33 @@ package object internals {
           }
       }
   }
+
+  def extractHeaders(
+      maybeHeaders: Option[Map[String, String]]
+  ): Option[Headers] =
+    maybeHeaders.map(h =>
+      Headers(h.toList.flatMap(parseSingleHeader).map(a => a: Header.ToRaw): _*)
+    )
+
+  // regex for comma not between quotes
+  private val commaNotBetweenQuotes = ",(?=([^\"]*\"[^\"]*\")*[^\"]*$)"
+
+  private def parseSingleHeader(
+      kv: (String, String)
+  ): List[(String, String)] = {
+    val key = kv._1
+    val values = kv._2.split(commaNotBetweenQuotes).toList
+    @tailrec
+    def loop(
```

Comment:
could this be replaced by a `map`: `values.map(value => key -> value)` ?

Reply:
lol , im so tired